Packs/ApiModules/Scripts/JSONFeedApiModule/JSONFeedApiModule.py

```diff
@@ -0,0 +1,163 @@
+from CommonServerPython import *
+''' IMPORTS '''
+from typing import List, Dict, Union
+import jmespath
+import urllib3
+
+# disable insecure warnings
+urllib3.disable_warnings()
+
+FEED_NAME: str
+
+
+class Client:
+    def __init__(self, url: str, credentials: Dict[str, str] = None,
+                 feed_name_to_config: Dict[str, dict] = None, source_name: str = 'JSON',
+                 extractor: str = '', indicator: str = 'indicator', fields: Union[List, str] = None,
+                 insecure: bool = False, cert_file: str = None, key_file: str = None, headers: dict = None, **_):
+        """
+        Implements class for miners of JSON feeds over http/https.
+        :param url: URL of the feed.
+        :param credentials:
+            username: username for BasicAuth authentication
+            password: password for BasicAuth authentication
+        :param extractor: JMESPath expression for extracting the indicators from
+        :param indicator: the JSON attribute to use as indicator. Default: indicator
+        :param source_name: feed source name
+        :param fields: list of JSON attributes to include in the indicator value.
+        If None no additional attributes will be extracted.
+        :param insecure: if *False* feed HTTPS server certificate will be verified
+        Hidden parameters:
+        :param: cert_file: client certificate
+        :param: key_file: private key of the client certificate
+        :param: headers: Header parameters are optional to specify a user-agent or an api-token
+        Example: headers = {'user-agent': 'my-app/0.0.1'} or Authorization: Bearer
+        (curl -H "Authorization: Bearer " "https://api-url.com/api/v1/iocs?first_seen_since=2016-1-1")
+         Example:
+            Example feed config:
+            'AMAZON': {
+                'url': 'https://ip-ranges.amazonaws.com/ip-ranges.json',
+                'extractor': "prefixes[?service=='AMAZON']",
+                'indicator': 'ip_prefix',
+                'fields': ['region', 'service']
+            }
+        """
+
+        self.source_name = source_name or 'JSON'
+        if feed_name_to_config:
+            self.feed_name_to_config = feed_name_to_config
+        else:
+            self.feed_name_to_config = {
+                self.source_name: {
+                    'url': url,
+                    'indicator': indicator or 'indicator',
+                    'extractor': extractor or '@',
```

Comment:
what does @ mean?

Reply:
add explanation here and for the indicator too

Reply:
it's explained in the docstring, should I add anything more there?